src/content/app/genome-browser/components/browser-track-config/useBrowserTrackConfig.ts

```diff
@@ -0,0 +1,170 @@
+  updateApplyToAll,
+  TrackType
+} from '../../state/track-config/trackConfigSlice';
+
+import analyticsTracking from 'src/services/analytics-service';
+import useGenomeBrowser from '../../hooks/useGenomeBrowser';
+
+const useBrowserTrackConfig = () => {
+  const applyToAllConfig = useSelector(getApplyToAllConfig);
+  const browserCogList = useSelector(getBrowserCogList);
+  const selectedCog = useSelector(getBrowserSelectedCog) || '';
+  const activeGenomeId = useSelector(getBrowserActiveGenomeId);
+  const shouldApplyToAll = applyToAllConfig.isSelected;
+  const shouldApplyToAllRef = useRef(shouldApplyToAll);
+
+  const selectedTrackConfigInfo = useSelector((state: RootState) =>
+    getTrackConfigForTrackId(state, selectedCog)
+  );
+
+  useEffect(() => {
+    shouldApplyToAllRef.current = shouldApplyToAll;
+  }, [shouldApplyToAll]);
+
+  const dispatch = useDispatch();
+
+  const { toggleTrackName, toggleTrackLabel } = useGenomeBrowser();
+
+  const updateTrackName = (isTrackNameShown: boolean) => {
+    if (!activeGenomeId) {
+      return;
+    }
+    if (shouldApplyToAllRef.current) {
+      Object.keys(browserCogList).forEach((trackId) => {
+        dispatch(
+          updateTrackConfigNames({
+            genomeId: activeGenomeId,
+            selectedCog: trackId,
+            isTrackNameShown
+          })
+        );
+        toggleTrackName({ trackId, shouldShowTrackName: isTrackNameShown });
+      });
+    } else {
+      dispatch(
+        updateTrackConfigNames({
+          genomeId: activeGenomeId,
+          selectedCog,
+          isTrackNameShown
+        })
+      );
+      toggleTrackName({
+        trackId: selectedCog,
+        shouldShowTrackName: isTrackNameShown
+      });
+    }
+
+    analyticsTracking.trackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'track_name_' + (isTrackNameShown ? 'on' : 'off')
+    });
+  };
+
+  const updateTrackLabel = (isTrackLabelShown: boolean) => {
+    if (!activeGenomeId) {
+      return;
+    }
+
+    if (shouldApplyToAllRef.current) {
+      Object.keys(browserCogList).forEach((trackId) => {
+        dispatch(
+          updateTrackConfigLabel({
+            genomeId: activeGenomeId,
+            selectedCog: trackId,
+            isTrackLabelShown
+          })
+        );
+        toggleTrackLabel({
+          trackId,
+          shouldShowFeatureLabel: isTrackLabelShown
+        });
+      });
+    } else {
+      dispatch(
+        updateTrackConfigLabel({
+          genomeId: activeGenomeId,
+          selectedCog,
+          isTrackLabelShown
+        })
+      );
+      toggleTrackLabel({
+        trackId: selectedCog,
+        shouldShowFeatureLabel: isTrackLabelShown
+      });
+    }
+
+    analyticsTracking.trackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'feature_label_' + (isTrackLabelShown ? 'on' : 'off')
+    });
+  };
+
+  const handleRadioChange = (value: OptionValue) => {
```

Comment:
This name is almost meaningless. The reader of the code will wonder, what radio. The radio is in a different file now, and ultimately, is just an implementation detail.

Reply:
✅